include/fobnail-attester/meta.h

```diff
@@ -3,39 +3,70 @@
 
 #include <stdint.h>
 
-/* Structure for serial number */
-#define SN_SIZE 64
-struct serial_number {
-    uint8_t size;
-    uint8_t number[SN_SIZE];
-};
+/* Structure for meta data for provisioning and attestation */
+struct meta_data {
+    uint8_t     header_version;
+    uint8_t     mac[6];         /* MAC address */
+    char        *serial_number;       /* Serial number */
+    char        *manufacturer;
+    char        *product_name;
+} __attribute__((packed));
 
-/* Enum for hash id */
-enum {
-    SHA1,
-    SHA256,
-    SHA512
-};
+#define META_H_VERSION  1
+
+#define SYS_FIRMWARE_DIR "/sys/firmware/dmi/tables"
+#define SYS_ENTRY_FILE SYS_FIRMWARE_DIR "/smbios_entry_point"
+#define SYS_TABLE_FILE SYS_FIRMWARE_DIR "/DMI"
 
-/* Structure for EK cert hash */
-struct EKcert_hash {
-    uint8_t id; /* specify what hash algo is currently used, could be equal SHA1, SHA256... */
-    union {
-        uint8_t sha1_hash[20];
-        uint8_t sha256_hash[32];
-        uint8_t sha512_hash[64];
-    } hash;
+struct dmi_header {
+    uint8_t  type;
+    uint8_t  length;
+    uint16_t handle;
+    uint8_t  *data;
```

Comment:
After this additional changes are needed: `{MRF,SN,PN}_OFFSET` have to be 4 bytes lower and initial `*bp` in `_dmi_string()` will also have to point at beginning of `dm` instead of `dm->data`.

Reply:
Ok, fixed.